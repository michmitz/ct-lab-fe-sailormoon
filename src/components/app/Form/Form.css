.hidden {
  display: none
}

.form-div {
  display: flex;
  justify-content: center;
}
