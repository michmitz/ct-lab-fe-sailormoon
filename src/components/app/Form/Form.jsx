import React from 'react';
import PropTypes from 'prop-types';
import './Form.css';

const Form = ({ sailorName, realName, description, attack, zodiacSign, imageUrl, onChange, onSubmit, successDisplay }) => (
  <div key="form-div-key" className="form-div">

      <form onSubmit={onSubmit} className="create-sailor-form">
        <div className="form-header">
          <h1>Create Your Own Sailor!</h1>
        </div>

        <div className="sailor-name-box">
          <label>Enter your home planet, star or moon:
            <input 
              type="text" 
              name="sailorName" 
              placeholder="ex. Deimos" 
              value={sailorName} 
              onChange={onChange} />
          </label>
        </div>

        <div className="real-name-box">
          <label>Enter your real name:
            <input 
              type="text" 
              name="realName" 
              placeholder="" 
              value={realName} 
              onChange={onChange} />
          </label>
        </div>

        <div className="description-box">
          <label>Give the fans a brief description of yourself:
            <input 
              type="text" 
              name="description" 
              placeholder="" 
              value={description} 
              onChange={onChange} />
          </label>
        </div>

        <div className="attack-box">
          <label>What's your number one "attack"?
            <input 
              type="text" 
              name="attack" 
              placeholder="ex. Dead Scream" 
              value={attack} 
              onChange={onChange} />
          </label>
        </div>

        <div className="zodiac-sign-box">
          <label>What's your zodiac sign?
            <input 
              type="text" 
              name="zodiacSign" 
              placeholder="" 
              value={zodiacSign} 
              onChange={onChange} />
          </label>
        </div>

        <div className="image-box">
          <label>Paste your image url here:
            <input 
              type="text" 
              name="imageUrl" 
              placeholder="" 
              value={imageUrl} 
              onChange={onChange} />
          </label>
        </div>

        <button data-testid="form-button">Submit</button>
      </form>

      <div data-testid="success-display">
        <p>{successDisplay}</p>
      </div>

  </div>
);

Form.propTypes = {
  sailorName: PropTypes.string.isRequired,
  realName: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  attack: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  successDisplay: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Form;