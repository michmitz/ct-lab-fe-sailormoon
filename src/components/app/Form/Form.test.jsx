import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Form from './Form';

describe('Form component', () => {
  afterEach(() => cleanup());

  it('renders Form', () => {
    const { asFragment } = render(<Form
      sailorName="Sailor Neptune"
      realName="Michiru Kaiou"
      description="Michiru is exactly what a fairy tale princess would be like."
      attack="Deep Submerge"
      zodiacSign="Pisces"
      imgUrl="https://michirusite/michiru.png"
      onChange={() => {}}
      onSubmit={() => {}}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
