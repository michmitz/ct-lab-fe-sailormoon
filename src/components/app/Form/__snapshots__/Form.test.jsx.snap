// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<DocumentFragment>
  <div
    class="form-div"
  >
    <form
      class="create-sailor-form"
    >
      <div
        class="form-header"
      >
        <h1>
          Create Your Own Sailor Moon Character!
        </h1>
      </div>
      <div
        class="sailor-name-box"
      >
        <label>
          Enter your home planet, star or moon:
          <input
            name="sailor-name"
            placeholder="ex. Deimos"
            type="text"
            value="Sailor Neptune"
          />
        </label>
      </div>
      <div
        class="real-name-box"
      >
        <label>
          Enter your real name:
          <input
            name="real-name"
            placeholder=""
            type="text"
            value="Michiru Kaiou"
          />
        </label>
      </div>
      <div
        class="description-box"
      >
        <label>
          Give the fans a brief description of yourself:
          <input
            name="description"
            placeholder=""
            type="text"
            value="Michiru is exactly what a fairy tale princess would be like."
          />
        </label>
      </div>
      <div
        class="attack-box"
      >
        <label>
          What's your number one "attack"?
          <input
            name="attack"
            placeholder="ex. Dead Scream"
            type="text"
            value="Deep Submerge"
          />
        </label>
      </div>
      <div
        class="zodiac-sign-box"
      >
        <label>
          What's your zodiac sign?
          <input
            name="zodiac-sign"
            placeholder=""
            type="text"
            value="Pisces"
          />
        </label>
      </div>
      <div
        class="image-box"
      >
        <label>
          Paste your image url here:
          <input
            name="image-url"
            placeholder=""
            type="text"
            value=""
          />
        </label>
      </div>
      <button
        data-testid="form-button"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
